### ISSUE TITLE
Search bar in Browser

### ISSUE THREAD

**Issue by Intuos5** [qgis/QGIS#44101]
**Feature description.**
A small quality of life improvement, It would be nice to have a searchbar in the Browser panel to quickly find particular files/ folders. It would be similar to the search bar in the Processing Toolbox.

**Comment by pathmapper** [comment]
@Intuos5 what about the filter button wich provides a search bar?

<img src="https://user-images.githubusercontent.com/20856381/125071835-1a9d7e00-e0ba-11eb-8734-375dafda43ec.png" width="400">

**Comment by Intuos5** [comment]
@pathmapper the filter only searches amongst visible items, ideally, you would be able to navigate to a folder that is nested deep within a drive by typing its name in the search box. This is thus different from the filter that is already there. Again, just like Windows Explorer's search, which searches from a given directory down to all files and folders based on the input string.

**Comment by pathmapper** [comment]
> you would be able to navigate to a folder that is nested deep within a drive by typing its name in the search box.

Looks like this is already the case if you opened/navigated to the specific nested folder once with QGIS browser.

**Comment by Intuos5** [comment]
Yes, if you have expanded the entire folder structure that is, but my request is specifically to search and to click on the result to navigate to a file/ particular folder without expanding the browser first.

**Comment by kannes** [comment]
I think this would need some mandatory preferences to setup as otherwise it would iterate through all drives, all potential network shares, all databases and schemata? So the user should rather explicitly pick the tree items to be included before getting a friendly phone call from IT.

**Comment by gioman** [comment]
I very much agree that as it is now, the browser filter is not very useful. I think that there is a lot of room for improvement here, for example there could be a way to tell QGIS that some specific folders and datasources (i.e. WMS, PostGIS connections) must be "open" on QGIS start, so that the browser filter would actually return results even if they were not opened manually.

Maybe a dialog similar to the "change data source" one would work well here.
